lib/reactor.cc

```diff
@@ -34,120 +33,132 @@ ReactorElement::ReactorElement(const std::string& name,
-      break;
-    case Type::Reaction:
-      container->register_reaction(reinterpret_cast<Reaction*>(this));
-      break;
-    case Type::Reactor:
-      container->register_reactor(reinterpret_cast<Reactor*>(this));
-      break;
-    default:
-      throw std::runtime_error("unexpected type");
+  case Type::Action:
+    container->register_action(reinterpret_cast<BaseAction *>(this)); //NOLINT
+    break;
+  case Type::Port:
+    container->register_port(reinterpret_cast<BasePort *>(this)); //NOLINT
+    break;
+  case Type::Reaction:
+    container->register_reaction(reinterpret_cast<Reaction *>(this)); //NOLINT
+    break;
+  case Type::Reactor:
+    container->register_reactor(reinterpret_cast<Reactor *>(this)); //NOLINT
+    break;
+  default:
+    throw std::runtime_error("unexpected type");
   }
 
-  std::stringstream ss;
-  ss << _container->fqn() << '.' << name;
-  _fqn = ss.str();
+  std::stringstream string_stream;
+  string_stream << container_->fqn() << '.' << name;
+  fqn_ = string_stream.str();
 }
 
-ReactorElement::ReactorElement(const std::string& name,
+ReactorElement::ReactorElement(const std::string &name,
                                ReactorElement::Type type,
-                               Environment* environment)
-    : _name(name), _fqn(name), _container(nullptr), _environment(environment) {
-  toggle_assert(environment != nullptr);
+                               Environment *environment)
+    : name_(name), fqn_(name), environment_(environment) {
+  reactor_assert(environment != nullptr);
   validate(type == Type::Reactor,
            "Only reactors can be owned by the environment!");
-  validate(this->_environment->phase() == Environment::Phase::Construction,
+  validate(environment_->phase() == Environment::Phase::Construction,
            "Reactor elements can only be created during construction phase!");
 }
 
-Reactor::Reactor(const std::string& name, Reactor* container)
+Reactor::Reactor(const std::string &name, Reactor *container)
     : ReactorElement(name, ReactorElement::Type::Reactor, container) {}
-Reactor::Reactor(const std::string& name, Environment* environment)
+Reactor::Reactor(const std::string &name, Environment *environment)
     : ReactorElement(name, ReactorElement::Type::Reactor, environment) {
   environment->register_reactor(this);
 }
 
-void Reactor::register_action([[maybe_unused]]BaseAction* action) {
-  toggle_assert(action != nullptr);
-  reactor::validate(this->environment()->phase() == Environment::Phase::Construction,
-           "Actions can only be registered during construction phase!");
-  toggle_assert(_actions.insert(action).second);
+void Reactor::register_action([[maybe_unused]] BaseAction *action) {
+  reactor_assert(action != nullptr);
+  reactor::validate(
+      this->environment()->phase() == Environment::Phase::Construction,
+      "Actions can only be registered during construction phase!");
+  reactor_assert(actions_.insert(action).second);
```

Comment:
I noticed a problem in the `register_*` methods while working on a bugfix on master. Why is the method argument labeled as `[[maybe_unused]]`? It should always be used and inserted into the according set.
I suspect that if assertions are off, the insertion is not executed. I wonder if this was different when we used the C style assert... Anyway, I think the right way of implementing this would be to always do the insertion and only assert the result.